prog_test.go

```diff
@@ -687,17 +687,12 @@ func TestProgramBindMap(t *testing.T) {
 }
 
 func TestProgramInstructions(t *testing.T) {
```

Comment:
Does comparing the Tag in our tests make sense in the first place if the output is so unpredictable?

Reply:
Might be worth an issue as well.

Reply:
We compare Tag instead of comparing Instructions. I don't think it does any harm, and when a check like this starts to fail, it usually leads to an interesting discovery, like in this case.

> Might be worth an issue as well.

What do you mean?

Reply:
Can we sanitise the Instructions ourselves? Should we store map ID in metadata?

Reply:
For the test case as it was before this patch; not really. Given the following prog:

```
	0: LdImmDW dst: r0 imm: -1
	2: LoadMapPtr dst: r1 fd: 6
	4: Mov32Imm dst: r0 imm: 0
	5: Exit
```

This is what the xlated insns look like:
```
	0: LdImmDW dst: r0 imm: -1
	2: LdImmDW dst: r1 imm: -103942818276608
	4: Mov32Imm dst: r0 imm: 0
	5: Exit
```

Src of instruction 2 is unset. Sanitizing the pointer is no problem, but no way to figure out this is a map load if I understand correctly.